dspace-api/src/main/java/org/dspace/identifier/doi/DOIOrganiser.java

```diff
@@ -799,4 +661,81 @@ private void setQuiet() {
         this.quiet = true;
     }
 
-}
\ No newline at end of file
+    private void batchProcess(List<DOI> doiList, DOIOperation operation, String operationName) {
+        int batchSize = getBatchSize();
+        int count = 0;
+
+        try {
+            for (DOI doiRow : doiList) {
+                DSpaceObject dso = doiRow.getDSpaceObject();
+                if (Constants.ITEM != dso.getType()) {
+                    LOG.warn("Skipping DOI " + doiRow.getDoi() + ": Currently DSpace supports DOIs for Items only.");
+                    continue;
+                }
+
+                try {
+                    // Reload the DSpaceObject to ensure it's attached to the current session
+                    dso = context.reloadEntity(dso);
+                    doiRow.setDSpaceObject(dso);  // Update the DOI object with the reloaded DSpaceObject
+
+                    operation.perform(context, provider, doiRow);
+                    count++;
+
+                    if (count % batchSize == 0) {
+                        context.commit();
+                        if (!quiet) {
+                            System.out.println("Processed " + count + " DOIs for " + operationName + ".");
+                        }
+                    }
+
+                    if (!quiet) {
+                        System.out.println("Successfully " + operationName + " DOI " + DOI.SCHEME + doiRow.getDoi() + ".");
+                    }
+                } catch (IdentifierException | SQLException ex) {
+                    LOG.error("Error " + operationName + " DOI " + doiRow.getDoi(), ex);
+                }
+
+                if (count % batchSize == 0) {
+                    context.uncacheEntity(doiRow);
+                }
+            }
+
+            if (count % batchSize != 0) {
+                context.commit();
+            }
+
+        } catch (SQLException ex) {
+            LOG.error("Database error during DOI " + operationName + " process", ex);
+            try {
+                context.rollback();
+            } catch (SQLException e) {
+                LOG.error("Error rolling back transaction", e);
+            }
+        }
+    }
+
+    public void update(List<DOI> doiList) {
+        batchProcess(doiList, (context, provider, doi) -> 
+            provider.updateMetadataOnline(context, doi.getDSpaceObject(), DOI.SCHEME + doi.getDoi()),
```

Comment:
It would be nice to extract `DOI.SCHEME + doi.getDoi()` with a more meaningful variable / method  since it's duplicated in other lines.